src/profiler/custom_op_profiler.h

```diff
@@ -0,0 +1,93 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+#ifndef MXNET_CUSTOM_OP_PROFILER_H_
+#define MXNET_CUSTOM_OP_PROFILER_H_
+
+#include <string>
+#include <unordered_set>
+#include <unordered_map>
+#include <thread>
+#include "./profiler.h"
+
+namespace mxnet {
+namespace profiler {
+
+using Tid = std::thread::id;
+using TaskPtr = std::shared_ptr<ProfileTask>;
+
+class CustomOpProfiler {
+public:
+    static CustomOpProfiler* Get() {
```

Comment:
You can make the constructor of the class as private. Why do you need to use a mutex ?

Reply:
Yeah I will make the constructor private. I think the use of mutex is to prevent two threads entering Get() at the same time and creating two instances. This is how class Profiler implements its Get()